src/Raven.Server/Web/RequestHandler.cs

```diff
@@ -353,12 +358,47 @@ private static void InvalidCountOfValues(string name)
             throw new ArgumentException($"Query string value '{name}' must appear exactly once");
         }
 
-        protected DisposableAction TrackRequestTime()
+        protected DisposableAction TrackRequestTime(bool alertThresholdExceeded = true)
         {
-            HttpContext.Response.Headers.Add(Constants.Headers.RequestTime, "1");
-            return null; // TODO [ppekrol] we cannot write Headers after response have started without buffering
-            //var sw = Stopwatch.StartNew();
-            //return new DisposableAction(() => HttpContext.Response.Headers.Add(Constants.Headers.RequestTime, sw.ElapsedMilliseconds.ToString(CultureInfo.InvariantCulture)));
+            var sw = Stopwatch.StartNew();
+            _context.HttpContext.Response.OnStarting(state => 
+            {
+                var httpContext = (HttpContext)state;
+                httpContext.Response.Headers.Add(Constants.Headers.RequestTime, sw.ElapsedMilliseconds.ToString());
+                return Task.FromResult(0);
+            }, _context.HttpContext);
+
+            return new DisposableAction(() =>
+            {
+                try
+                {
+                    if (alertThresholdExceeded)
+                    {
+                        var threshold = _context.RavenServer.Configuration.Core.TooLongRequestThreshold.AsTimeSpan;
+                        if (threshold.TotalMilliseconds >= sw.ElapsedMilliseconds)
```

Comment:
This is not a good idea in this manner.
For example, we routinely have requests that last 15 seconds, any request to create an index that is waiting for it, for example.

I think that the default should be 30 seconds.